edison: Add acpi sample tables

htot opened the pull request:
Tables authored by Andy Shevchenko<andriy.shevchenko@linux.intel.com>
Retrieved from https://edison.internet-share.com/wiki/ACPI

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
edison: Add acpi sample tables

Tables authored by Andy Shevchenko<andriy.shevchenko@linux.intel.com>
Retrieved from https://edison.internet-share.com/wiki/ACPI

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot commented:
I added Andy's sample tables, but haven't yet figured out how to build them. Also I'm not sure the way the compiled tables are deployed is the best way. What do you think?

andy-shev committed:
Merge pull request #6 from alext-mkrs/update-acpica-20180105

acpica: update to 20180105

alext-mkrs commented:
I'm looking into this one. Building tables should be easy enough and the tables themselves would probaly need to be verfied? (not sure what's their status - tested and solid, just a first stab, etc). I'll post back when I have them building.

alext-mkrs commented:
Ok, looks like it's not going to be that easy. We can technically _build_ them (as that's just a matter of running them through the `iasl`), however to build the full image (and deploy those tables there), it's first necessary to make `meta-acpi` support Edison as the `MACHINE`. And that suddenly becomes noticeably more work and AFAIU that mostly would duplicate what we're doing in https://github.com/htot/meta-intel-edison. On the other hand, `meta-acpi` could serve as a foundation, a very "vanilla" way of building an ACPI-supporting image for Edison, just like it does for Galileo and others, so I can see value in getting it to support Edison.

Let's maybe clarify this with @andy-shev and @westeri - what do you folks think of adding Edison to the supported `MACHINE`s list for `meta-acpi` and how hard do you think that would be?

htot commented:
I have them building, and created a do_install function so they get packaged. The packages then go into the initramfs. I am hoping this will work (as kernel documentation says it must be an uncompressed cpio), otoh the same guy that added acpi-overlay through cpio added through initramfs one year later. I'm just not sure it that works for these type of tables.

htot committed:
acpi-tables: build package

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot commented:
Push what I have now. To add it to initramfs you need:
`PACKAGE_INSTALL += "kernel-module-mmc-block kernel-module-mmc-core kernel-module-sdhci kernel-module-sdhci-acpi kernel-module-sdhci-pci acpi-tables"`
in core-image-minimal-initramfs.bbappend

alext-mkrs commented:
Yes, that would work within the `meta-intel-edison` context, but not `meta-acpi` one (as the latter doesn't support Edison as target machine). So for `meta-acpi` you changes would pretty much be invisible, as Edison example tables won't be compiled and the ne `do_install()` function shouldn't break anything as well. I'd say - if @andy-shev has no problems with that, go for a PR for the main `meta-acpi` repo.

htot commented:
Apparently the arduino.asl is an 'ssdt', whatever that is. So in the kernel documentation this applies: https://github.com/htot/linux/blob/eds/Documentation/acpi/ssdt-overlays.txt.
I found also this, which suggests we should be able to do this from initramsfs as well: https://www.spinics.net/lists/linux-initramfs/msg03341.html but it doesn't seem to work.
I will try using configfs now.

htot commented:
I added a kernel fragment to enable configfs:
`CONFIG_ACPI_CONFIGFS=y`
and built kernel new and reboot.

Then copied the aml files to ~ manually. Then:
```
cd /sys/kernel/config/acpi/table/
mkdir arduino
cat ~/arduino.aml > arduino/aml
mkdir spidev
cat ~/spidev.aml > spidev/aml
```

This indeed loads the aml tables.
```
root@edison:/sys/kernel/config/acpi/table# dmesg | grep -i acpi
..
[  310.463210] ACPI: Host-directed Dynamic ACPI Table Load:
[  310.468607] ACPI: SSDT 0xFFFFA22A7C68D000 00085F (v05        ARDUINO  00000001 INTL 20180105)
[  310.491823] ACPI: Executed 1 blocks of module-level executable AML code
[  311.697165]       bus-0135 bus_get_status        : Device [ACPI] status [0000000f]
[  456.118899] ACPI: Host-directed Dynamic ACPI Table Load:
[  456.124283] ACPI: SSDT 0xFFFFA22A64081900 0000A6 (v05        SPIDEV   00000001 INTL 20180105)
[  456.144620] ACPI: Executed 1 blocks of module-level executable AML code
[  457.081003]       bus-0135 bus_get_status        : Device [ACPI] status [0000000f]
```

alext-mkrs commented:
> I found also this, which suggests we should be able to do this from initramsfs as well: https://www.spinics.net/lists/linux-initramfs/msg03341.html but it doesn't seem to work.

I'd check if the initramfs generated indeed contains the compiled tables and in the expected directory, that would the first suspect IMHO.

htot commented:
Actually there is a slew of messages in dmesg after the manual acpi table load including errors that mean nothing to me: [ACPI: Host-directed Dynamic ACPI Table Load](https://docs.google.com/document/d/1SFeR8WRA-QO2jUE04dfXAu1ryf7IYcnSALtgoWbVhW8/edit?usp=sharing). Maybe @andy-shev knows what that is.

htot commented:
Boot with no sd card and debugshell into initramfs:
```
Cannot find /dev/mmcblk1 or no /sbin/init present , dropping to a shell
sh: can't access tty; job control turned off
/ # ls /
bin/     dev/     init     lib/     run/     sys/     var/
boot/    etc/     kernel/  proc/    sbin/    usr/
/ # ls /kernel/firmware/acpi/
arduino.aml  spidev.aml
```

htot commented:
Interesting, before loading the acpi tables I get:
```
root@edison:~# i2cdetect -r 1
WARNING! This program can confuse your I2C bus, cause data loss and worse!
I will probe file /dev/i2c-1 using read byte commands.
I will probe address range 0x03-0x77.
Continue? [Y/n] 
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: 20 21 22 23 -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --                         
```
and after:
```
root@edison:/sys/kernel/config/acpi/table# i2cdetect -r 1
WARNING! This program can confuse your I2C bus, cause data loss and worse!
I will probe file /dev/i2c-1 using read byte commands.
I will probe address range 0x03-0x77.
Continue? [Y/n] 
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          -- -- -- -- -- -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: UU UU UU UU -- -- -- -- -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --                         
```
So it looks like something is working.

alext-mkrs commented:
> Boot with no sd card and debugshell into initramfs: <...>

Hmm, that'sinteresting then - apparently the files are there. The next step would probably then be to check who/how specifically loads those, i.e. which code path in kernel would do that.

> Actually there is a slew of messages in dmesg after the manual acpi table load including errors that mean nothing to me: ACPI: Host-directed Dynamic ACPI Table Load.

I'm no expert either, but this looks like those tables are incomplete or something, so it complains at not finding some of the contexts. But yeah, let's wait for what Andy has to say.

>Interesting, before loading the acpi tables I get: <...> So it looks like something is working.

It certainly looks so, thumbs up :)

htot commented:
Kernel doc specifically say to use uncompressed cpio, copied in front of compressed initrd cpio. Nowhere says how to handle a kernel with bundled initramfs.
As a initramfs loads earlier than initrd I would have thought it should work.

alext-mkrs commented:
I think unless @andy-shev knows right away that loading ACPI tables from initramfs (as opposed to initrd) is not supposed to work, the only definitive way of clarifying this would be to analyze the kernel early boot code related to the initrd case + the initramfs code, to see what's the difference. I can do that, but that'd take some time.

htot commented:
I check kernel code and as far as I understand the cpio containing acpi tables is indeed loaded earlier than initramfs, which in turn is loaded earlier than initrd. If that sticks, then I should be able to have u-boot just load the cpio and zboot. Should be easy enough to test this (hopefully Wed evening). Just to see if there is a difference compared to using configfs.

But I like configfs, loading appropriate acpi tables could be done by mraa. What do you think?

andy-shev commented:
Guys, *initrd* is not in use for more than 5 years. I doubt anyone is actually do *initrd* anymore. The name (like in kernel command line) is kept the same for sake of compatibility.

htot commented:
initrd doesn't seem to be easy to use, but afaikt it's still supported and a different mechanism than initramfs.
This https://github.com/htot/linux/blob/eds/Documentation/acpi/initrd_table_override.txt in particular line 79, only discusses initrd but I think the takeaway is that the cpio with acpi files need to be loaded first and must be uncompressed regardless of what is contained in the 2nd. In our case there is no second. Or actually bitbake builds: 1) kernel, 2) initramfs cpio, 3) kernel with built-in initramfs and now also 4) cpio with acpi. Up to now we use 3)

andy-shev commented:
@htot 
While first sentence is correct, the second one is not fully true. As I said **name is kept for sake of compatibility**. Thus, further you did a right conclusion, i.e. *the cpio with acpi files need to be loaded first and must be uncompressed regardless of what is contained in the 2nd*.

alext-mkrs commented:
So we can try to concatenate the initramfs cpio and the acpi table one in the same way the docs instruct to combine acpi cpio and initrd. It may be the case, as Andy suggests, that "initrd" name is used, but initramfs is also accounted for.

htot commented:
It appears: not needed to concatenate. Initramfs already built into the kernel and is loaded relatively late. Just tested. I already had:
```
bootargs_usbboot=debugshell=10 tty1 console=ttyS2,115200n8 root=/dev/sda rootfstype=ext4 systemd.unit=multi-user.target hardware_id=00
```
Now added:
```
root@edison:~# fw_setenv boot_acpiboot 'zboot 0x100000 0 0x3000000 0x1000000'
root@edison:~# fw_setenv load_acpiboot 'load mmc 0:9 0x100000 bzImage-initramfs-usb; load mmc 0:9 0x3000000 acpi-tables.cpio'
root@edison:~# fw_setenv acpiboot 'setenv bootargs ${acpi} ${bootargs_usbboot}; run load_acpiboot; run boot_acpiboot'
```
This results to  (see the interesting last 2 lines):
```
=> run acpiboot
reading bzImage-initramfs-usb
10293120 bytes read in 454 ms (21.6 MiB/s)
reading acpi-tables.cpio
3584 bytes read in 15 ms (232.4 KiB/s)
Valid Boot Flag
Setup Size = 0x00004000
Magic signature found
Using boot protocol version 2.0d
Linux kernel version 4.15.0-rc8-edison-acpi-standard (oe-user@oe-host) #2 SMP Thu Jan 18 23:27:01 CET 2018
Building boot_params at 0x00090000
Loading bzImage at address 100000 (10276736 bytes)
Magic signature found
Initial RAM disk at linear address 0x03000000, size 16777216 bytes
Kernel command line: "acpi.debug_layer=0xffff056d acpi.debug_level=0x0000200f debugshell=10 tty1 console=ttyS2,115200n8 root=/dev/sda rootfstype=ext4 systemd.unit=multi-user.target hardware_id=00"


RAMDISK: [mem 0x03000000-0x03ffffff]
ACPI: SSDT ACPI table found in initrd [kernel/firmware/acpi/arduino.aml][0x85f]
ACPI: SSDT ACPI table found in initrd [kernel/firmware/acpi/spidev.aml][0xa6]
...
modified: [mem 0x00000000377fd000-0x00000000377fd904] ACPI data
ACPI: Early table checksum verification disabled
ACPI: RSDP 0x00000000000E4500 000024 (v02 U-BOOT)
ACPI: XSDT 0x00000000000E45E0 00003C (v01 U-BOOT U-BOOTBL 20180120 INTL 00000000)
ACPI: FACP 0x00000000000E4FC0 0000F4 (v06 U-BOOT U-BOOTBL 20180120 INTL 00000000)
ACPI: DSDT 0x00000000000E4780 000732 (v02 U-BOOT U-BOOTBL 00010000 INTL 20180105)
ACPI: APIC 0x00000000000E50C0 000048 (v04 U-BOOT U-BOOTBL 20180120 INTL 00000000)
ACPI: MCFG 0x00000000000E5110 00003C (v01 U-BOOT U-BOOTBL 20180120 INTL 00000000)
ACPI: Table Upgrade: install [SSDT-      - ARDUINO]
ACPI: SSDT 0x00000000377FD000 00085F (v05        ARDUINO  00000001 INTL 20180105)
ACPI: Table Upgrade: install [SSDT-      -  SPIDEV]
ACPI: SSDT 0x00000000377FD85F 0000A6 (v05        SPIDEV   00000001 INTL 20180105)
...
ACPI: Core revision 20170831
 dsmethod-0188 ds_detect_named_opcode: Method serialized [_OSC] (ptrval) - [CreateDWordField] (008A)
...
Initializing Namespace objects:
Table [DSDT: U-BOOTBL] (id 01) -   64 Objects with  11 Devices,   2 Regions,   24 Methods (3/21/1 Serial/Non/Cvt)
Table [SSDT: ARDUINO ] (id 02) -   45 Objects with   6 Devices,   0 Regions,   11 Methods (0/11/0 Serial/Non/Cvt)
Table [SSDT: SPIDEV  ] (id 03) -    4 Objects with   1 Devices,   0 Regions,    0 Methods (0/0/0 Serial/Non/Cvt)
ACPI: 3 ACPI AML tables successfully acquired and loaded
APIC: Switch to symmetric I/O mode setup
...
Unpacking initramfs...
Initramfs unpacking failed: junk in compressed archive
```
It boot fine nevertheless. I guess it finds no compressed cpio after the acpi.cpio and falls back to the one built into the kernel.

htot commented:
For the rest the results seem to be the same as with loading through configfs. I would think:

1. For developing loading acpi through configfs is easiest, could be done with a script, through systemd or from mraa depending on the peripherals you need. This roughly translates to scenarios booting from usb/sd card - suggestions?
2. For final products (not sure) loading the acpi.cpio and closing configfs is maybe better. Roughly when booting from internal flash image.

So the recipe needs to be fixed up to not copy files to initramfs, maybe same location on the rootfs? Or /usr/share?

andy-shev commented:
(Can you enable issue tracker here? Or we can move it to meta-acpi issuer tracker)

WRT logical split between Edison layer(s) and meta-acpi. I discussed a bit with Mika and we conclude that the best approach would be following
- meta-acpi layer provides an *edison* machine
- meta-acpi provides set of samples of ASL code (let's start with what I have for Edison/Arduino)
- selecting *edison* machine may require dependencies to the Edison specific layer(s)
- meta-acpi **might** have an additional feature to build U-Boot (not Edison specific, means just upstream version, no other patches included)
- meta-acpi **should not** have Edison specific or other very machine specific stuff, **except** ACPI tables, everything should be taken from upstream repositories of certain projects.

htot commented:
@andy-shev your 5 bullets are already taken care of here https://github.com/htot/meta-acpi/tree/eds/recipes-bsp/acpi-tables

We already had edison machine, thus by adding the sample asl in the created edison directory everything is already in place to get the uncompressed cpio built (thanks Mika!).

However I added a few lines in the acpi-tables.bb https://github.com/htot/meta-acpi/commit/f3ed062c40a0e258d2072868cbaf8e0fb576dad0#diff-9dac3372107d01699bdfb9bec784f797 . 

Originally the cpio is built and deployed (in one step), I added an install step (this means install asl's into a certain directory _on the build system_, further by listing the files in the FILES variable a deb or ipk or rpm is created (_automatically packaged_) and copied to another directory in the build system. This is normal and doesn't hurt any other machine. The cpio is still deployed (copied to the deploy directory on the build system) as before.

The deb is only installed to the target image when specifically added in the targets image recipe.

Currently for eds I add the deb package (wrongly) to the initramfs recipe, so the AML's are installed into the initramfs the same as ko modules we require. And with the current state of the kernel these are ignored. But neither initramfs or target image recipe are part of meta-acpi.

meta-acpi builds other stuff that we don't use (kernel) or might use later, but I don't think we should add u-boot, without patches it should be just coming from [openembedded-core](http://layers.openembedded.org/layerindex/branch/master/recipes/?q=u-boot)

So it's already as you want. There is only one thing to decide that affects the packaging into debs: do we want one deb with all the selected AML from the ACPI_TABLES variable? Or do we want one deb per AML? I think the former.

For the latter we need a package split step, currently we are not doing that. Right now deb and cpio contain exactly the same AML's. Potentially the deb could contain a script and systemd unit that automatically loads the AML's through configfs. Separate loading of AML's can be done manually.

Nevertheless, as I tested yesterday loading the cpio from u-boot works fine for eds (and nothing changed for the existing machines).

alext-mkrs commented:
@andy-shev, thanks, that makes sense. I haven't explicitly disabled issue tracker in this fork (maybe that's GH default with forks?), but in any case I'd propose to have any meaningful records or discussions in the main repo, to have all in one place, if that's what you mean.

@htot, meta-acpi doesn't explicitly support _edison_ machine as of now. You are able to build it with _meta-intel-edison_, because it would provide such support, but not without it. For Galileo, Minnow and Joule there's such support and that's what I was talking about - deciding on the approach for Edison.

alext-mkrs commented:
@htot, as for the loading, I think your experiment just confirmed the initrd mechanism described in the docs, works :) However I had something else in mind - initramfs is actualy built as a cpio, which is appended to the kernel binary by simple concatenation. So my wild guess is that if prepending initrd binary with acpi-containing cpio works, then it very well may be the case that doing the same with initramfs cpio would also work. I plan to test it his weekend probably.

htot commented:
Alright. I had a quick look myself and concluded (too soon?) that as we have the INITRAMFS_IMAGE_BUNDLE = 1, the initramfs cpio is passed to the kernel build system as uncompressed cpio for inclusion. If so, I don't see how we can get the two concatenated and bundled. But this part of YP could be better documented and the code is (too me) almost incomprehensible.

htot commented:
@alext-mkrs I found that GH copies the issues setting when forking.

On supporting edison you mean there is no edison.conf? You are right, that one is provided by meta-intel-edison. I don't know how YP intends this, I would expect the conf and machine to be part of the bsp. And if another layer supports the machine if would check for the machine. But I'm really no expert on this, so if we need to change something that's fine. For now I think it would be best to add only minimal extensions to meta-acpi until edison stabilizes.

andy-shev commented:
@alext-mkrs , yeah, this is how GH works. Nevertheless, feel free to open an issue for creating a minimum support for Edison machine in meta-acpi.

andy-shev committed:
galileo: Use *.asli extention for included files

Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>

andy-shev committed:
linux-kernel: Bump Linux kernel to v4.14

Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>

andy-shev committed:
linux-kernel: Refresh defconfig

Signed-off-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>

andy-shev commented:
I have updated meta-acpi a bit, so, what I'm expecting now is just a minimum support for Edison there:

1. Introduce *edison* machine
2. Adding Edison/Arduino archive (better, perhaps, one patch per example. like: base + spidev + AD/C + ...)
3. (as next step) Introducing U-Boot recipe.

Does it make sense?

alext-mkrs commented:
Yep, it sure does. Yes, I think base + specific single example, + maybe a full set makes the most sense and easier to understand.

htot commented:
It's like @alext-mkrs wrote above: what you propose is to redo (or move) stuff. Right now (as a remnant from the original meta-intel-edison) we have edison.conf here: https://github.com/htot/meta-intel-edison/blob/master/meta-intel-edison-bsp/conf/machine/edison.conf and the machine _edison_ here: https://github.com/htot/meta-intel-edison/blob/pyro64-acpi/local.conf
Further kernel + u-boot are here: https://github.com/htot/meta-intel-edison/tree/pyro64-acpi/meta-intel-edison-bsp

I think the question is: is meta-acpi a bsp for all the mentioned boards (i.e. replacement for meta-intel-edison-bsp)? Or is it a layer that adds acpi to these boards bsp's (included into meta-intel-edison as I have now)? From the description, I think it is the latter. But then, why is it providing a kernel?

If we just move these things to meta-acpi everything will remain working and meta-acpi will become a bsp supporting multiple boards.

For me: it's all ok. I don't know how the layers for the other boards are structured. Do they have a common bsp? Where can that be found? And when they are not using meta-acpi, do they have a non-acpi alternative bsp?

htot committed:
acpi-tables: add script and systemd service

This installs a script acpi-tables-load that can be manually run to load all acpi tables in /kernel/firmware/acpi
on a configfs enabled kernel. The script assumes configfs is mounted on /sys/kernel/config as is the default.
Further a systemd service acpi-tables-load.service is installed that runs the script . To run once manually,
use 'systemctl start acpi-tables-load', to automatically run after boot 'systemctl enable acpi-tables-load'

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
acpi-tables: update

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
acpi-tables: make recipe understand compile options through ACPI_FEATURES in "uart_2w uart_4w spi i2c"

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
acpi-tables: fix warnings

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
acpi-tables: REVERTME: set TRI_STATE_ALL before loading tables

It is essential to set TRI_STATE_ALL low before and high after setting the external muxes
(hardware guide Document Number: 331191-008 March 2017).

When this makes it into the acpi table this commit should be reverted.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

alext-mkrs commented:
Just to close on this one, which was still bugging me :)

> Alright. I had a quick look myself and concluded (too soon?) that as we have the INITRAMFS_IMAGE_BUNDLE = 1, the initramfs cpio is passed to the kernel build system as uncompressed cpio for inclusion. If so, I don't see how we can get the two concatenated and bundled. But this part of YP could be better documented and the code is (too me) almost incomprehensible.

I've tested the way I had in mind - prepending the compressed initramfs cpio with the `acpi-tables.cpio` before it got built into the kernel (by hacking a couple of recipes a little - there're no ready-made means of doing that using standard Yocto classes).

That combination _does_ boot (so it apparently is able to parse such a "sandwich" of initramfs cpios), however I see nothing regarding that SSDT table or specific devices listed in Edison examples (`arduino.asl/spidev.asl`) being initialized in ACPI-related kernel logs (even with the `acpi.debug_level/layer` bumped up) - there seem to be only those from U-Boot-supplied tables. That's sort of strange as I would expect it to recognize them, as the load/override mechanism at play is essentially the same as with the "initrd override", so maybe we're missing something else here. @andy-shev, what do you think? Here are logs with varying levels of verbosity in case you can spot anything: https://pastebin.com/ZngaqH42

I did that using the latest set of commits from https://github.com/htot/meta-intel-edison/pull/16, so @htot you may consider it tested - ACPI-enabled kernel 4.15 builds and boots fine.

htot commented:
So, it loads ramfs cpio, but ignores the acpi cpio? And the acpi cpio is the first and is not compressed? Strange.
But looking in the logs it is indeed not loaded. Normally you would get:
ACPI: **3** ACPI AML tables successfully acquired and loaded (https://edison.internet-share.com/wiki/ACPI). In your log is only 1.

alext-mkrs commented:
Yep, that's how it looks and yes  that's strange as it theoretically should be able to parse the acpi table cpio to be able to load the initramfs one. So my guess is - it throws then away for some reason. All things I could think of (table type, format, content) seem to beok, so there must be something else, thus my call out to Andy, maybe he could spot something.

htot commented:
My understanding was that the yocto recipe for initramfs unpacks the cpio in a certain directory and then the kernel build compresses again and links into the kernel image. If that is correct that you always end up with a compressed cpio?

alext-mkrs commented:
Yes, that's the case, but there's another option with using uncompressed cpio (no unpacking then) and I've hacked recipes to use that. I think I'll check it further a bit later.

htot commented:
Ah, well it will be interesting to see how you are doing that. I wait patiently:-)

andy-shev commented:
@htot 
Yes, the cpio:s have to be unpacked and prepend the main initramfs.

htot committed:
acpi-tables: don't toggle TRI_STATE_ALL if i2c drivers not yet loaded

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
acpi-tables: switch to using libgpiod when DISTRO_FEATURES has acpi (#1)

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

andy-shev commented:
I have got a notification against this PR and I didn't find why. Hmm...

htot commented:
Ah. This is strange. A very old PR. We created an organization and moved the relevant repo's there. So when I pushed to that it keeps getting added to this PR and since you are a participant you got notified too.

I'll close this one for now to prevent that from happening again.

htot commented:
But if you would like to join you are very welcome!
